packages/core-data/src/hooks/use-entity-record-query.js

```diff
@@ -0,0 +1,50 @@
+/**
+ * WordPress dependencies
+ */
+import { useState, useCallback } from '@wordpress/element';
+import { useSelect, __experimentalResolveSelect } from '@wordpress/data';
+
+export default function useEntityRecordQuery(
+	kind,
+	name,
+	{ initialPage = 1, perPage = 10, ...query } = {}
```

Comment:
If this hook is targetted specifically for pagination, I wonder if we should separate the pagination config from the query config.

Reply:
otherwise, people will have to provide their own initialPage, perPage values even if they just want to provide a query.

Reply:
Why? Those two properties have default values in the destructuring so it would be fine.

Reply:
Righ! got confused for a second.